src/ActionBuilder/Product/AddVariants.php

```diff
@@ -49,20 +53,19 @@ public function createUpdateActions(
             return [];
         }
 
-        // Filter out removed variants
-        $changedValue = array_filter($changedValue);
+        list (, $catalog) = $this->getLastFoundMatch();
 
-        return $this->getAddVariantActions($changedValue);
+        return $this->getAddVariantActions($sourceObject->getMasterData()->{'get' . ucfirst($catalog)}()->getVariants());
```

Comment:
What happens with a new added master variant? This won't return by `getVariants`

Reply:
New MasterVariants will be handled by the ChangeMasterVariantAction.